src/vs/workbench/contrib/positronConsole/browser/positronConsoleActions.ts

```diff
@@ -608,6 +643,151 @@ export function registerPositronConsoleActions() {
 		}
 	});
 
+	/**
+	 * Register the "Execute Code From Beginning To Current Line" action.
+	 * This action executes all the code in the editor to the current line without moving the cursor.
+	 */
+	registerAction2(class extends Action2 {
+		constructor() {
+			super({
+				id: PositronConsoleCommandId.ExecuteCodeBeforeCursor,
+				title: {
+					value: localize('workbench.action.positronConsole.executeCodeBeforeCursor', 'Execute Code From Beginning To Current Line'),
+					original: 'Execute Code From Beginning To Current Line'
+				},
+				f1: true,
+				category,
+				precondition: ContextKeyExpr.and(
+					EditorContextKeys.editorTextFocus,
+					NOTEBOOK_EDITOR_FOCUSED.toNegated(),
+					POSITRON_NOTEBOOK_CELL_EDITOR_FOCUSED.toNegated()
+				),
+				keybinding: {
+					weight: KeybindingWeight.WorkbenchContrib,
+					primary: KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.Home
+				}
+			});
+		}
+
+		/**
+		 * Runs action.
+		 * @param accessor The services accessor.
+		 * @param opts Options for code execution
+		 *   - selectionMode: When set to 'beforeCursor', executes all code before the cursor. When set to 'afterCursor', executes all code after the cursor.
+		 *   - allowIncomplete: Optionally, should incomplete statements be accepted? If `undefined`, treated as `false`.
+		 *   - languageId: Optionally, a language override for the code to execute. If `undefined`, the language of the active text editor is used.
+		 *   - mode: Optionally, the code execution mode for a language runtime. If `undefined` fallbacks to `Interactive`.
+		 *   - errorBehavior: Optionally, the error behavior for a language runtime. If `undefined` fallbacks to `Continue`.
```

Comment:
We don't need any of these `opts` for now, but I was thinking of including them to pipe through to the new helper function, in case it is useful for extension developers? We've run into a similar situation before. Happy to remove them instead.